The ZNG reader can panic

To reproduce:

```
> git checkout origin/jamii-fuzz-vng-repro
> cd vng
> go test -run=FuzzVngRoundtripBytes/2287ef37feae2d65
panic: runtime error: slice bounds out of range [:-1]

goroutine 24 [running]:
github.com/brimdata/zed/pkg/peeker.(*Reader).Peek(0x8cb700?, 0xc00009e910?)
	/home/jamie/zed/pkg/peeker/reader.go:77 +0x110
github.com/brimdata/zed/pkg/peeker.(*Reader).Read(0xc00009e910, 0xffffffffffffffff)
	/home/jamie/zed/pkg/peeker/reader.go:81 +0x27
github.com/brimdata/zed/zio/zngio.(*parser).readCompressedFrame(0xc00028a518, 0x27?)
	/home/jamie/zed/zio/zngio/parser.go:186 +0x1a8
github.com/brimdata/zed/zio/zngio.(*parser).decodeTypes(0xc00028a518, 0x0?)
	/home/jamie/zed/zio/zngio/parser.go:61 +0x2e
github.com/brimdata/zed/zio/zngio.(*parser).read(0xc00028a518)
	/home/jamie/zed/zio/zngio/parser.go:45 +0x1a8
github.com/brimdata/zed/zio/zngio.(*scanner).start.func1()
	/home/jamie/zed/zio/zngio/scanner.go:118 +0x95
created by github.com/brimdata/zed/zio/zngio.(*scanner).start
	/home/jamie/zed/zio/zngio/scanner.go:104 +0x14a
exit status 2
FAIL	github.com/brimdata/zed/vng	0.005s
```

This is not terrible, but it might cause format detection to fail if the zng reader panics on non-zng data?

--- a/pkg/peeker/reader.go
+++ b/pkg/peeker/reader.go
@@ -60,6 +60,9 @@ func (r *Reader) fill(need int) error {
 }
 
 func (r *Reader) Peek(n int) ([]byte, error) {
+	if n < 0 {
+		return nil, errors.New("peeker: negative length")
+	}
 	if len(r.cursor) == 0 && r.eof {
 		return nil, io.EOF
 	}

--- a/pkg/peeker/reader_test.go
+++ b/pkg/peeker/reader_test.go
@@ -17,3 +17,11 @@ func TestEOFOnEmptyFill(t *testing.T) {
 	assert.ErrorIs(t, err, io.EOF)
 	assert.Len(t, n, 0)
 }
+
+func TestNegativeLength(t *testing.T) {
+	p := NewReader(nil, 0, 0)
+	_, err := p.Read(-1)
+	assert.ErrorContains(t, err, "peeker: negative length")
+	_, err = p.Peek(-1)
+	assert.ErrorContains(t, err, "peeker: negative length")
+}

